include/onnxruntime/core/framework/ml_value.h

```diff
@@ -69,4 +69,31 @@ class MLValue {
   MLDataType type_{nullptr};
   FencePtr fence_;
 };
+
+struct MLValueAllocationParameters {
+  MLValueAllocationParameters() = default;
+  MLValueAllocationParameters(const TensorShape* shape)
+      : tensor_shape{shape} {}
+
+  const TensorShape& GetTensorShape() const {
+    static const TensorShape s_empty_tensor_shape;
+    return tensor_shape != nullptr ? *tensor_shape : s_empty_tensor_shape;
+  }
+
+ private:
+  const TensorShape* tensor_shape{};
+  // todo: is there any parameter needed for ml types?
```

Comment:
as said, looks to me there's no, if agreed, remove this struct.

Reply:
Totally agree. It adds no value.